### ISSUE TITLE
TaggedValue and TaggedValueList React components

### ISSUE THREAD

**Issue by vikasrohit** [appirio-tech/react-components#53]
**Requirements**
- [x] Should be styled as shown in [specks](https://drive.google.com/drive/u/0/folders/0B1LbmKnex41fMkVfWFhSUlR3eGM). Please see the bonus item in Prize tooltip
- [x] TaggedValue should ask for tag color, label of tag and corresponding value as properties
- [x] TaggedValue should render the number of times the tagged value is to be shown, as part of the component.
- [x] ~~TaggedValueList should ask for color scheme, for now use two `assorted` and `default`. `default` would use two colors (mentioned in the specks), one for TaggedValue items where the count is greater than 1 and another for TaggedValue items where count is 1. `assorted` would use different colors from a fixed palette.~~
Removed this requirement, as @amolskh implemented a better way to control the color schemes and delegated the logic for default theme to TaggedValue component.
- [x] TaggedValueList should support two different layouts (configurable via property). One layout would allow wrapping of items when all items takes more space than the parent container and another would use horizontal scroll.

**Comment by amolskh** [comment]
I am taking this one 

Regards

**Comment by vikasrohit** [comment]
:+1:  Go ahead.

**Comment by amolskh** [comment]
Could you help me locate component in that specks.. I can see description for Progressbar but not this

**Comment by vikasrohit** [comment]
From requirements:
>>Please see the bonus item in Prize tooltip<<

Basically, it is a generic version of what we are building (or in fact you have already built it) as PrizeItem with only exception that it allows the count (e.g. x 5) to be displayed as well. I know we can do that with your PrizeItem code, however, that requires the 'x 5' to be included in the `subText` property of the component. On the other side, TaggedValue component would allow user to specify count as separate property and it would be responsibility of the component to render the 'x {count}'. That means we may use this component in PrizeItem component.

**Comment by amolskh** [comment]
If look and feel wise the component is same how about having optional   field called count in PrizeItem if optional field is present add 'x {count}' else render 'subText'  Does that work?

**Comment by vikasrohit** [comment]
Indirectly, we would be doing the same. But we need this component as generic option for all other such components (e.g. PrizeItem). Further, TaggedValueList would also has option to adjust the layout such that it can wrap the elements or put a mobile friendly horizontal scroll.
In summary, it would be more generic than PrizeItem. We can do the reverse i.e. use this component in PrizeItem once you are done with it.

**Comment by vikasrohit** [comment]
@amolskh Please let me know if you still have any confusion about these components and Prize components.

**Comment by amolskh** [comment]
No I think i am good as of now.. Will try to given you code by Today EOD

**Comment by amolskh** [comment]
If I understand correctly font color and background color code are supposed to be dynamic right?
How is this expected to be achieved ? inline css in .jsx or any other way

**Comment by vikasrohit** [comment]
We can use classes which user can provide to the component (as array to TaggedValueList and as string to TaggedValue). And we can apply those classes to the component elements. If you see any problem with that, we are open to use any other approach which seems do the work with best practices. This is the place where we need your skill to get it done in better fashion.

If we settle down on my suggested approach, TaggedValue would take one css class which can specify background and font color both. On the other hand, TaggedValueList would take array of css classes and apply them based on layout param.

**Comment by amolskh** [comment]
"TaggedValue should render the number of times the tagged value is to be shown, as part of the component."
As of now I can think of something like this 
<TaggedValue title="1" subText="$10000" styleClass="colors"/>

**Comment by vikasrohit** [comment]
Sorry, @amolskh however, I Didn't get you.

**Comment by amolskh** [comment]
"TaggedValue should render the number of times the tagged value is to be shown, as part of the component." This line is mentioned in the requrirement
 TaggedValue title="1" subText="$10000" styleClass="colors"
along with above values I will be passing some sort of count ? if count is 2 I will be rendering componenet 2 times? or is the requirement trying to refer (subText X Count)
Its not very clear what  exactly is expected from that particular line

**Comment by vikasrohit** [comment]
Sorry for the confusion.
Yes, you will be passing the count as prop and 

> or is the requirement trying to refer (subText X Count)

This is correct.

**Comment by amolskh** [comment]
Could you elaborate for me more on 'TaggedValueList' component color scheme assorted and default..Also logic if count is 1 or more. Its not very clear TaggedValue We have already provided the styleclass which mentions styling of individual TaggedValue correct?

I can think of providing one more style element for TaggedValue List which can be css styles as per requirement

**Comment by vikasrohit** [comment]
Let me try again

TaggedValue component would take style class (let say tagStyle) as argument which would be used to style the tag. Now when TaggedValueList component uses the TaggedValue component, it need to pass that tagStyle for each TaggedValue item. For this we have two options default or assorted. With default, we would use default styles which is to have TaggedValue items with count being 1 as purple colored, and items with count being greater than 1 as blue colored. With assorted style, we would use provided (via property of TaggedValueList component) palette, which is provided as array of css classes, to style individual TaggedValue item. If the palette has less colors then the items in the TaggedValueList component, we would duplicate colors otherwise we can pick(you can use simple logic for picking the class which is to pick css classes from the palette in order i.e. for item[i] use class palette[i]) any one css style from the palette.

Hope this helps.

**Comment by amolskh** [comment]
I have checked in the code . Please review and let me know your thoughts

**Comment by vikasrohit** [comment]
@amolskh Final fixes:
- [ ] Styles for TaggedValueList component should be in separate file
- [ ] Can we restrict the values of layout property using enums syntax `React.PropTypes.oneOf(['News', 'Photos'])`
